gcp/service/dataplex : 10 policies with 10 test cases

puniaruj opened the pull request:
Contribute to dataplex. Include 10 policys and 10 test case.
You can find my policy in /policies/gcp/dataplex.
And test case is in /inputs/gcp/dataplex.

puniaruj committed:
All 10 policies with Documentation

puniaruj committed:
Resolved Conflicts

visal-dam reviewed (changes requested):
Your policies do not meet the expected format. Please do not create your own functions, as it makes your policy incompatible with the current set. This is the job of the helpers, which ensures that the same static checking format is used project-wide. If you were interested in Rego development, you were advised to contact the leadership team.

Your policy logic also seems to assessed the changed resources, rather than relevant configs; changed resources also contain previous configurations, which are irrelevant. Your policies also only access surface-level attributes and do not support nested attributed - this is why we made the helpers, templates, and guides.

Your policies cannot be accepted. Please rework them all according to the current standard and documentation.

puniaruj committed:
Fixed two policies according to feedback

puniaruj committed:
changed all 10 policies according to reviewer

I changed surface-level attributes nested attributed , used helpers template and tested according to that

puniaruj committed:
Update .gitignore

puniaruj committed:
Update .gitignore

puniaruj committed:
Update .gitignore

puniaruj committed:
.

visal-dam commented on policies/gcp/dataplex/google_dataplex_lake/metastore/policy.rego:
Whitelisting null, so a metastore config must not be present? This policy contradicts the remedy statement.

puniaruj replied:
Thanks, Jerry. You are right — currently it whitelists null. My intention was to require that a metastore config must be present. I will update the policy so that the whitelist values reflect the valid configurations instead of null. This way the policy and remedy will align.

visal-dam commented on policies/gcp/_helpers/helpers.rego:
We do not want any changes to the helpers.rego, insignificant or otherwise, from policy PRs. Add helpers.rego to your gitignore.

puniaruj replied:
Got it. I’ll revert any changes to helpers.rego and add it to .gitignore so it’s never included in my PRs again.

visal-dam commented on policies/gcp/dataplex/google_dataplex_asset/json_options/policy.rego:
In this case, if a resource has BOTH UTF-8 and json_options.disable... false, then it is compliant. If it has UTF-8 but json_options... is true, it is still compliant as only condition c2 is triggered. If it has UTF-16, but json_options... is false, then it is still compliant as only c1 is triggered. If it has BOTH UTF-16 and json_options.... is true, then it is NON-compliant as both c1 and c2 are triggered. Is this what you intended? Please confirm. If so, explain why.

puniaruj replied:
Yes, that is the intended logic. The policy only marks resources as non-compliant if both UTF-16 encoding and json_options.disable set to true occur together, since that combination introduces parsing and security risks. Any other single condition is acceptable.

visal-dam commented on policies/gcp/dataplex/google_dataplex_asset/labels/policy.rego:
policy contradicts remedy. also, we don't consider labels to be security-relevant in the PDE, but you can keep these policies for the time being.

visal-dam commented on policies/gcp/dataplex/google_dataplex_zone/labels/policy.rego:
Policy looks good. Your attribute paths are consistent if the assessed value is a pure dictionary in your json file. Please confirm.

puniaruj replied:
Confirmed — labels are represented as a pure dictionary in the Terraform plan output, so the attribute path ["labels"] is correct and consistent.

visal-dam commented on policies/gcp/dataplex/google_dataplex_zone/type/policy.rego:
Is there a reason why "Dataplex Zones must have type set to RAW or CURATED"? Please explain in your description.

visal-dam reviewed (changes requested):
Minor changes. Please also confirm some of the questions raised. Also, please ensure that your next PR does not contain any other files requested from you, namely the inputs and policies of YOUR SERVICE ONLY. Update your gitignore accordingly. Thanks.

puniaruj committed:
Revert helpers.rego and ignore in git

puniaruj committed:
fixed all the files
